paywall/src/hooks/components/Web3Service.js

```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types'
+import React, { createContext } from 'react'
+
+import web3Service from '../../services/web3Service'
+
+export const Web3ServiceContext = createContext()
+
+export default function Web3Service({ children }) {
```

Comment:
I think it is going to be confusing to have many `Web3Service`. Since this is not a service but a component, could we just name this `Web3Component`?